metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py

```diff
@@ -643,13 +652,28 @@ def get_foreign_key_metadata(
             fk_dict["name"], foreign_fields, source_fields, foreign_dataset
         )
 
+    def init_data_reader(self, inspector: Inspector) -> Optional[DataReader]:
```

Comment:
init implies that it's called as part of `__init__`, which isn't the case

```suggestion
    def make_data_reader(self, inspector: Inspector) -> Optional[DataReader]:
```